### PR TITLE
Arc - Replace hard coded String with class reference

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Nice.

@mkouba FYI, the rationale is that it makes the Jakarta migration easier.

**Review by mkouba — APPROVED** [review]

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java [hunk 1]
```diff
@@ -71,7 +71,8 @@
 
     void createSharedAnnotationLiteral(ClassOutput classOutput, Key key, Literal literal, Set<String> existingClasses) {
         // Ljavax/enterprise/util/AnnotationLiteral<Lcom/foo/MyQualifier;>;Lcom/foo/MyQualifier;
-        String signature = String.format("Ljavax/enterprise/util/AnnotationLiteral<L%1$s;>;L%1$s;",
+        String signature = String.format("L%1$s<L%2$s;>;L%2$s;",
+                AnnotationLiteral.class.getName().replace('.', '/'),
```
Comment (gastaldi) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java:75]: FTR in JDK 12+ `java.lang.Class#descriptorString` can be called instead

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java:75]: But we're still on JDK 11, or?

Comment (gastaldi) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java:75]: Yes, still in JDK11, just a curiosity 😉

**Pull request merged**